pkg/registry/types.go

```diff
@@ -114,6 +114,71 @@ type Annotations struct {
 	DefaultChannelName string `json:"operators.operatorframework.io.bundle.channel.default.v1" yaml:"operators.operatorframework.io.bundle.channel.default.v1"`
 }
 
+// DependenciesFile holds dependency information about a bundle
+type DependenciesFile struct {
+	// Dependencies is a list of dependencies for a given bundle
+	Dependencies []Dependency `json:"dependencies" yaml:"dependencies"`
+}
+
+// Dependencies is a list of dependencies for a given bundle
+type Dependency struct {
+	// The type of dependency. It can be `olm.package` for operator-version based
+	// dependency or `olm.gvk` for gvk based dependency. This field is required.
+	Type string `json:"type" yaml:"type"`
+
+	// The name of dependency such as 'etcd'
+	Name string `json:"packageName" yaml:"packageName"`
+
+	// The group of GVK based dependency
+	Group string `json:"group" yaml:"group"`
+
+	// The kind of GVK based dependency
+	Kind string `json:"kind" yaml:"kind"`
+
+	// The version of dependency in semver format
+	Version string `json:"version" yaml:"version"`
```

Comment:
Similar comments to the what I mentioned for protobuf. We either want a `value` bag to put things in, or we want a set of different types.

To highlight the problem: Version here is `semver` if `type=package` but `string` (specifically `kube apiversion string`) if `type=gvk`